### DISCUSSION TITLE
Hello I want to push this local repository to my remote repository but I have a problem

### DISCUSSION THREAD

**Discussion by safepre** [community/community#43033] [New to GitHub] [answered]
[](url)fatal: 'master' does not appear to be a git repository fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.
here is the image I got this error right when I try to push it to the remote repository
How can I fix this issue? I want the origin and master to sync so I can start pushing my code again

![image](https://user-images.githubusercontent.com/97003952/210164439-2279b41d-8038-4062-a295-4dadcaa5136c.png)

**Comment by airtower-luna** [comment, marked as answer]
What's the *exact* command you're using? The message

> fatal: 'master' does not appear to be a git repository fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

indicates the way you specify the repository to push to is probably wrong. If you want to update `origin/master` to the state of your local `master` you need to use:

```
git push origin master
```

`origin` is the name of the remote repository, `master` the ref (branch or tag) to push. Remote repositories have names because you can have any number of them, and typing a name is much faster than typing the full URL.

See also: [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)
